enhancements/cluster-logging/loki-storage.md

```diff
@@ -0,0 +1,628 @@
+    // requirement for single redudancy and auto-compaction.
+    OneXMedium LokiStackSizeType = "1x.medium"
+)
+
+```
+
+The redundancy policy declares the requirement how many Loki ingester replicas should process each log stream in parallel. This does not always result in redudant stores on the object storage. It is a safety measure to allow failover of ingester replicas because of unexpected disruption (e.g. node rescheduling, cluster upgrades, etc.). In turn `ZeroRedundancy` can result in data loss.
+
+__Note:__ Redundancy of the persistent object storage is configured by the administator per bucket based on the object storage provider policies available (e.g. multi-zone/region buckets).
+
+```go
+
+// RedundancyPolicyType defines the policy for the log stream redundancy.
+//
+// +kubebuilder:validation:Enum=ZeroRedundancy;SingleRedundancy
+type RedundancyPolicyType string
+
+const (
+    // ZeroRedundancy defines that each log stream should be processed
+    // at maximum by one ingester instance.
+    ZeroRedundancy RedundancyPolicyType = "ZeroRedundancy"
+
+    // SingleRedundancy defines that each log stream should be processed
+    // at minimum by two separate ingester instances.
+    SingleRedundancy RedundancyPolicyType = "SingleRedundancy"
+)
+```
+
+#### Mandatory: Loki deployment specification
+
+The mandatory fields of a `LokiStack` custom resource declaration are:
+- `Size`: Selection of one of the supported t-shirt size formats (see [Types and constants](#types-and-constants)) to reconcile the `CPU/Memory Requests`, `CPU/Memory Limits`, `Replicas` and `PodDisruptionBudgets` per Loki component.
+- `Storage`: A value of `ObjectStorageSpec` to declare the S3 compatible-API `URL` endpoint and a `Secret` located in `openshift-logging` by a human operator. If the `Secret` provides a key-value pair for `endpoint` then the `URL` is overridden.
+- `RedundancyPolicy`: A selection of the supported redundancy policies (see [Types and constants](#types-and-constants)). All selections except `ZeroRedundancy` require a valid `Size` selection or the deployment will not proceed and reported as degraded state in the `LokiStackStatus`, i.e:
+  - `SingleRedundancy` requires at minimum `1x.medium`.
+
+The optional parts are `Limits` (see discussion in section [Optional: Global/Per-Tenant Limits](#optional-globalper-tenant-limits)) and `Template` (see details in section [Optional: Loki component configuration](#optional-Loki-component-configuration)).
+
+__Note__: The `Template` field if left empty will be populated by the [loki-operator](https://github.com/viaq/loki-operator) on the first reconciliation by translating the `Size` selection. This automatic update is to inform the user about the opinionated defaults for each supported scale out beyond documentation. The user can use the `Template` field to override these defaults.
+
+```go
+// ObjectStorageSecretSpec is a secret reference containing name only, no namespace.
```

Comment:
I think it's Kubernetes convention to put secrets in the namespace with the things that will use them. I would not suggest changing this to allow the secrets to exist anywhere. If the idea is to prevent access then RBAC is the solution.

Reply:
I suggest for the start to keep things simple and the operator should live and deploy its operands on `openshift-logging`. To keep RBAC management to a minimum, we should start simple and inform the administrator to create a secret in `openshift-logging`.

Reply:
Agreed. Needs some research to justify the extra flexibility, and maybe even the research isn't justified if this is common practice and nobody has complained to date.